packages/axios-extension/src/abap/abap-service-provider.ts

```diff
@@ -182,52 +171,29 @@ export class AbapServiceProvider extends ServiceProvider implements AbapServiceP
     }
 
     /**
-     *
-     * @param packageName Package name for deployment
-     * @param appName Fiori project name for deployment. A new project that has not been deployed before is also allowed
-     * @returns array of transports id's
+     * Retrieve singleton instance of AdtService subclass to serve the specific ADT request query.
+     * @example
+     * ```ts
+     * const transportRequestSerivce = abapServiceProvider.getAdtService<TransportRequestService>(TransportRequestService);
+     * ```
+     * @param adtServiceSubclass Subclass of class AdtService, type is specified by using AdtService class constructor signature.
+     * @returns Subclass type of class AdtService
      */
-    public async getTransportRequests(packageName: string, appName: string): Promise<TransportRequest[]> {
-        let serviceSchema: AdtCollection;
-        try {
-            serviceSchema = await this.getAdtCatalogService().getServiceDefinition(
-                AdtServiceConfigs[AdtServiceName.TransportChecks]
-            );
-        } catch {
-            // Service not available on target ABAP backend version, return empty setting config
-            return [];
-        }
-
-        if (!serviceSchema || !serviceSchema.href) {
-            return [];
-        }
-        const urlPath = serviceSchema.href;
-        const acceptHeaders = {
-            headers: {
-                Accept: 'application/vnd.sap.as+xml; dataname=com.sap.adt.transport.service.checkData',
-                'content-type':
-                    'application/vnd.sap.as+xml; charset=UTF-8; dataname=com.sap.adt.transport.service.checkData'
+    public async getAdtService<T extends AdtService>(adtServiceSubclass: typeof AdtService): Promise<T> {
```

Comment:
```suggestion
    public async getAdtService<T extends AdtService>(adtServiceSubclass: typeof AdtService): Promise<T |null> {
```
Does that make more sense?

Reply:
Updated